sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/ClientEncryptionPolicy.java

```diff
@@ -128,11 +160,20 @@ private void validateClientEncryptionIncludedPath(ClientEncryptionIncludedPath c
         }
 
         if (clientEncryptionIncludedPath.getPath().charAt(0) != '/'
-            || clientEncryptionIncludedPath.getPath().lastIndexOf('/') != 0
-            || clientEncryptionIncludedPath.getPath().substring(1).equals("id")) {
+            || clientEncryptionIncludedPath.getPath().lastIndexOf('/') != 0) {
             throw new IllegalArgumentException("Invalid path " + clientEncryptionIncludedPath.getPath());
         }
 
+        if (clientEncryptionIncludedPath.getPath().substring(1).equals("id")) {
+            if (policyFormatVersion < 2) {
+                throw new IllegalArgumentException(String.format("Path %s cannot be encrypted with policyFormatVersion %s.", clientEncryptionIncludedPath.getPath(), policyFormatVersion));
+            }
+
+            if (!clientEncryptionIncludedPath.getEncryptionType().equals("Deterministic")) {
```

Comment:
Same here, we should not use string literals.

Reply:
I have fixed it now.